assets/src/domain/eventEditor/services/apollo/mutations/useReorderEntities.ts

```diff
@@ -0,0 +1,127 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+import { clone, reverse } from 'ramda';
+import { useDebouncedCallback } from 'use-debounce';
+
+import { EntityId } from '@appServices/apollo/types';
+import { getGuids } from '@appServices/predicates';
+import { Datetime, Ticket } from '../types';
+
+type Entity = Datetime | Ticket;
+
+interface ReorderEntitiesProps {
+	entityType: 'DATETIME' | 'TICKET';
+}
+
+interface CallbackArgs<E extends Entity> {
+	allEntities: Array<E>;
+	filteredEntities: Array<E>;
+	newIndex: number;
+	oldIndex: number;
+	updateEntityList: (updatedEntities: Array<E>) => void;
+}
+
+type SortCallback<E extends Entity> = (args: CallbackArgs<E>) => void;
+
+const REORDER_ENTITIES = gql`
+	mutation REORDER_ENTITIES($input: ReorderEspressoEntitiesInput!) {
+		reorderEspressoEntities(input: $input) {
+			ok
+		}
+	}
+`;
+
+interface ReorderEntities<E extends Entity> {
+	cancel: VoidFunction;
+	done: VoidFunction;
+	sortEntities: SortCallback<E>;
+}
+
+const useReorderEntities = <E extends Entity>({ entityType }: ReorderEntitiesProps): ReorderEntities<E> => {
+	const [allEntityGuids, setAllEntityGuids] = useState<Array<EntityId>>([]);
+	useEffect(() => {
+		if (allEntityGuids.length) {
+			done();
+		}
+	}, [allEntityGuids.join(':')]);
```

Comment:
this should be extracted based on:

![image](https://user-images.githubusercontent.com/1477453/80110339-400e9c80-8587-11ea-9386-9278ff6dd559.png)

Reply:
that warning is displayed when `react-hooks/exhaustive-deps` rule is set to `warn` or `error`, in fact I tried to enable that in one of my previous PRs but got too many warnings and Travis wasn't happy with that, but I think we should enable that rule in a separate PR and fix all the warnings or do smth else

Reply:
Done